seacrowd/sea_datasets/miracl/miracl.py

```diff
@@ -0,0 +1,287 @@
+
+_LANGUAGE_MAP = {
+    "id": "Thai", 
+    "th": "Indonesian"
+}
+
+_URLS = {_DATASETNAME: {lang: {} for lang in _LANGUAGE_MAP}}
+
+for lang in _LANGUAGE_MAP:
+    _URLS[_DATASETNAME][lang]['train'] = [
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/topics/topics.miracl-v1.0-{lang}-train.tsv',
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/qrels/qrels.miracl-v1.0-{lang}-train.tsv',
+    ]
+
+    _URLS[_DATASETNAME][lang]['dev'] = [
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/topics/topics.miracl-v1.0-{lang}-dev.tsv',
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/qrels/qrels.miracl-v1.0-{lang}-dev.tsv',
+    ]
+    
+    _URLS[_DATASETNAME][lang]['testB'] =[
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/topics/topics.miracl-v1.0-{lang}-test-b.tsv',
+    ]
+
+    _URLS[_DATASETNAME][lang]['testA'] = [
+        f'https://huggingface.co/datasets/miracl/miracl/resolve/main/miracl-v1.0-{lang}/topics/topics.miracl-v1.0-{lang}-test-a.tsv',
+    ]
+
+
+_SUPPORTED_TASKS = [Tasks.TEXT_RETRIEVAL]  # example: [Tasks.TRANSLATION, Tasks.NAMED_ENTITY_RECOGNITION, Tasks.RELATION_EXTRACTION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+def load_topic(fn):
+    qid2topic = {}
+    with open(fn, encoding="utf-8") as f:
+        for line in f:
+            qid, topic = line.strip().split('\t')
+            qid2topic[qid] = topic
+    return qid2topic
+
+
+def load_qrels(fn):
+    if fn is None:
+        return None
+
+    qrels = defaultdict(dict)
+    with open(fn, encoding="utf-8") as f:
+        for line in f:
+            qid, _, docid, rel = line.strip().split('\t')
+            qrels[qid][docid] = int(rel)
+    return qrels
+
+def seacrowd_config_constructor(lang, schema, version):
+    if lang not in _LANGUAGE_MAP:
+        raise ValueError(f"Invalid lang {lang}")
+
+    if schema != "source" and schema != "seacrowd_pairs":
+        raise ValueError(f"Invalid schema: {schema}")
+
+    return SEACrowdConfig(
+        name="miracl_{lang}_{schema}".format(lang=lang, schema=schema),
+        version=datasets.Version(version),
+        description="MIRACL {schema} schema for {lang} language".format(lang=_LANGUAGE_MAP[lang], schema=schema),
+        schema=schema,
+        subset_id="miracl",
+    )
+
+class Miracl(datasets.GeneratorBasedBuilder):
+    """MIRACL is a multilingual retrieval dataset that focuses on search across 18 different languages."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [seacrowd_config_constructor(lang, "source", _SOURCE_VERSION) for lang in _LANGUAGE_MAP] + [seacrowd_config_constructor(lang, "seacrowd_pairs", _SEACROWD_VERSION) for lang in _LANGUAGE_MAP]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
```

Comment:
running using the default config name `miracl_source` will not work here because it is not available,

we can either change the default to `f"{_DATASETNAME}_id_source"` or implement `miracl_source` to use all languages

Reply:
Is there an option to not have any default config name? If none, we can do `f"{_DATASETNAME}_id_source"`

In my opinion, it may not be ideal to do all languages because it's not faithful to the source schema.

Reply:
For this, I do agree w/ @danjohnvelasco. It's okay not to implement any default config for multilingual datasets (incl. this one).

Reply:
I set `DEFAULT_CONFIG_NAME = None` for this dataset. I pushed the changes.